core/src/test/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpServiceResponseConverterTest.java

```diff
@@ -168,6 +181,111 @@ public HttpData httpData() {
                     return ImmutableMap.of("a", "¥");
                 }
             });
+
+            sb.annotatedService("/custom-response", new Object() {
+                @Get("/expect-specified-status")
+                @StatusCode(202)
+                public Void expectSpecifiedStatus() {
+                    // Will send '202 Accepted' because a user specified it with @StatusCode annotation.
+                    return null;
+                }
+
+                @Get("/expect-no-content")
+                public void expectNoContent() {
+                    // Will send '204 No Content' because the return type is 'void'.
+                    return;
+                }
+
+                @Get("/expect-ok")
+                public Object expectOk() {
+                    // Will send '200 Ok' because there is no @StatusCode annotation, that means
+                    // '200 Ok' will be used by default because the return type is not a 'void' or 'Void'.
+                    return null;
+                }
+
+                @Get("/expect-specified-no-content")
+                @StatusCode(204)
+                public HttpResult<Object> expectSpecifiedNoContent() {
```

Comment:
I'm curious what would happen if a user defines methods like:

```java
public HttpResult<?> foo() { ... }
public <T> HttpResult<T> bar() { ... }
```

Will they work as expected?

Reply:
Added methods for those cases. Thanks! :-)